src/Core/Form/IdentifiableObject/DataHandler/OrderAddressFormDataHandler.php

```diff
@@ -0,0 +1,133 @@
+
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\Address\Command\EditOrderAddressCommand;
+use PrestaShop\PrestaShop\Core\Domain\Country\Exception\CountryConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Order\Exception\InvalidAddressTypeException;
+use PrestaShop\PrestaShop\Core\Domain\Order\Exception\OrderException;
+use PrestaShop\PrestaShop\Core\Domain\State\Exception\StateConstraintException;
+
+class OrderAddressFormDataHandler implements FormDataHandlerInterface
+{
+    /**
+     * @var CommandBusInterface
+     */
+    private $commandBus;
+
+    /**
+     * @param CommandBusInterface $commandBus
+     */
+    public function __construct(CommandBusInterface $commandBus)
+    {
+        $this->commandBus = $commandBus;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(array $data)
+    {
+        // Not used for creation, only edition
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws CountryConstraintException
+     * @throws StateConstraintException
+     * @throws InvalidAddressTypeException
+     * @throws OrderException
+     */
+    public function update($orderId, array $data)
+    {
+        $editAddressCommand = new EditOrderAddressCommand($orderId, $data['address_type']);
+
+        if (null !== $data['alias']) {
+            $editAddressCommand->setAddressAlias($data['alias']);
+        }
+
+        if (null !== $data['first_name']) {
+            $editAddressCommand->setFirstName($data['first_name']);
+        }
+
+        if (null !== $data['last_name']) {
+            $editAddressCommand->setLastName($data['last_name']);
+        }
+
+        if (null !== $data['address1']) {
+            $editAddressCommand->setAddress($data['address1']);
+        }
+
+        if (null !== $data['city']) {
+            $editAddressCommand->setCity($data['city']);
+        }
+
+        if (null !== $data['id_country']) {
+            $editAddressCommand->setCountryId((int) $data['id_country']);
+        }
+
+        if (null !== $data['postcode']) {
+            $editAddressCommand->setPostCode($data['postcode']);
+        }
+
+        if (null !== $data['dni']) {
+            $editAddressCommand->setDni($data['dni']);
+        }
+
+        if (null !== $data['company']) {
+            $editAddressCommand->setCompany($data['company']);
+        }
+
+        if (null !== $data['vat_number']) {
+            $editAddressCommand->setVatNumber($data['vat_number']);
+        }
+
+        if (null !== $data['address2']) {
+            $editAddressCommand->setAddress2($data['address2']);
+        }
+
+        if (null !== $data['id_state']) {
+            $editAddressCommand->setStateId((int) $data['id_state']);
+        }
+
+        if (null !== $data['phone']) {
+            $editAddressCommand->setHomePhone($data['phone']);
+        }
+
+        if (isset($data['phone_mobile']) && null !== $data['phone_mobile']) {
```

Comment:
```suggestion
        if (isset($data['phone_mobile'])) {
```
Why doing the same check? :sweat_smile:

Reply:
It is a dumb copy/paste from `AddressFormDataHandler` 😛
I think the isset is necessary because in some cases the field is not present, but then the latter part is useless indeed, I'll remove it